build.gradle

```diff
@@ -87,8 +87,23 @@ subprojects {
           }
         }
       }
+      repositories {
+        maven {
+          name = 'localTest'
+          url = "$rootProject.buildDir/local-test-repo"
+        }
+      }
     }
   }
+  // Make nebula and shadow play nice together, Gradle doesn't like it how one jar task replaces the output of another
+  // if both are part of published artifacts
+  project.afterEvaluate {
+    if (plugins.hasPlugin(MavenPublishPlugin) && plugins.hasPlugin(ShadowPlugin)) {
```

Comment:
Is this something that should be in BuildPlugin?

Reply:
I don't have a string preference. I'm thinking build plugin will be implemented in Java and perhaps the bulk of it split into smaller focused classes. In this respect small scripts and glue could live and would be easier to write in the groovy script.

Reply:
It seems wrong to me to have the shadow plugin and publishing be something we support via BuildPlugin, yet require additional hacks within our build files.